includes/class-wcpdf-install.php

```diff
@@ -378,6 +378,12 @@ protected function upgrade( $installed_version ) {
 			global $wpdb;
 			$documents = WPO_WCPDF()->documents->get_documents( 'all' );
 			foreach ( $documents as $document ) {
+				if ( ! function_exists( 'WPO_WCPDF_Pro' ) && $document->get_type() != 'invoice' ) {
```

Comment:
this shouldn't be limited to the existence of `WPO_WCPDF_Pro` the issue affects any custom documents for which no numbers have been generated yet. Instead of checking `{$wpdb->prefix}wcpdf_{$document->slug}_number` it should check for `{$number_store->table_name}`, which is the value that is used for the query after all!

Couldn't you do this right before this line?
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2c393f2c4888b3beabd111189530f1c51e8921d5/includes/class-wcpdf-install.php#L391

Reply:
No, the critical error comes from the `get_sequential_number_store()` invocation, which triggers the Packing Slip table creation, which in fact it happens but the `$EZSQL_ERROR` still catches the error don't know why:

![Captura de ecrã de 2022-02-23 09-52-38](https://user-images.githubusercontent.com/533273/155296085-a8aef80f-549f-4f8e-82c1-49af5e588a2e.png)

Reply:
I see - this appears to be similar to a `DESCRIBE` error we have encountered before, which was caused by WP itself and should have been fixed here: https://core.trac.wordpress.org/changeset/51582

Since there's no way around that from our end, your approach is probably the next best alternative, if you strip the part referring to `WPO_WCPDF_Pro`.

Reply:
I wonder though, could we silence errors in this function?
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/9b8dd26b79780871e16c5ed58b77648b4750b681/includes/documents/class-wcpdf-sequential-number-store.php#L195-L200

Reply:
```php
// suppress errors
$suppress_errors = $wpdb->suppress_errors();

// do stuff

// restore errors
$wpdb->suppress_errors( $suppress_errors );
```

Reply:
Can you explain why?

Reply:
@alexmigf did you see the above comments?

Reply:
> Can you explain why?

Because I think that's where the error originates?

Reply:
> I still got the same error using the `suppress_errors()` btw

Even when you wrap it around the `get_sequential_number_store()` call?

Reply:
I don't think it is because:

* The query is different, one starts with DESCRIBE and the other SHOW TABLES
* I have reset the `$EZSQL_ERROR` global variable after the `store_name_exists()` invocation and I still got the same error

Reply:
Yes, in `dbDelta()`: https://github.com/WordPress/WordPress/blob/f4683efbc96410ff1c3f99801080000398a4ddd9/wp-admin/includes/upgrade.php#L2776

Reply:
and that's a scenario where we _do_ want to catch errors, so we shouldn't disable it there...

Reply:
Exactly!